azure-pipelines.yml

```diff
@@ -0,0 +1,83 @@
+# File: azure-pipelines.yml
+
+trigger:
+  batch: true
+  branches:
+    include:
+      - main
+      - release/*
+
+resources:
+  repositories:
+    - repository: niveristand-custom-device-build-tools
+      type: github
+      ref: main
+      endpoint: ni (2)
+      name: ni/niveristand-custom-device-build-tools
+# TEMPORARY, include fake custom device to use templates until migrating the templates into build tools makes sense
+    - repository: imitation-cd
+      type: github
+      ref: main
+      endpoint: papowerNI
+      name: papowerNI/imitation-cd
+
+pool:
+  name: custom-device-temp
+  demands:
+    - agent.os -equals Windows_NT
+
+# --------- Build Parameters here ---------
+parameters:
+# Custom Device Name
+  - name: customDevice
+    type: string
+    default: niveristand-slsc-eds-custom-device
+# LabVIEW Versions to Build
+  - name: lvVersions
+    type: object
+    default:
+      - '2020'
+      - '2021'
+      - '2023'
+# Custom Device Build Steps
+  - name: buildSteps
+    type: object
+    default:
+      # (project location, LabVIEW CLI Build operation, Target, Build Specification Name)
+      - ['Source\SLSC EDS.lvproj', 'ExecuteBuildSpec', 'My Computer', 'Configuration Release']
+      - ['Source\SLSC EDS.lvproj', 'ExecuteBuildSpecAllTargets', '', 'Engine Release']
+# Build Output Location
+  - name: buildOutputLocation
+    type: string
+    default: 'Built'
+# Post-build Archive Location
+  - name: archiveLocation
+    type: string
+    default: '\\nirvana\temp\papower\Measurements\VeriStandAddons\slsc_eds_custom_device'
+
+# -----------------------------------------
+
+# TEMPORARY: template files reference imitation-cd instead of the build tools repo
+stages:
+  - stage: build
+    jobs:
+      - ${{ each lvVersion in parameters.lvVersions }}:
+        - job:
+          displayName: Build ${{ parameters.customDevice }} for LabVIEW ${{ lvVersion }}
+          steps:
+            - template: azure-templates\pre-job-steps.yml@imitation-cd # Configure variables, check out repos, Clear cache
+              parameters:
+                customDevice: ${{ parameters.customDevice }}
+                lvVersion:    ${{ lvVersion }}
+            - ${{ each buildStep in parameters.buildSteps }}:
```

Comment:
These `${{ each ... }}` blocks are just a shorthand for creating the jobs, right? 

In a more complex repo that might want to do different things for each LabVIEW version (e.g. SEECD with different projects building with different bitnesses), we could manually expand the jobs and steps?

Reply:
Yes, you can add additional steps before/after the template if you want, and make those steps conditional to lvVersion, assuming that those variables are available in the each loop and not defined within the template.